static/compiler.js

```diff
@@ -585,52 +587,56 @@ define(function (require) {
             }
         }
         var currentWord = this.outputEditor.getModel().getWordAtPosition(e.target.position);
-        var numericToolTip = currentWord && currentWord.word ? getNumericToolTip(currentWord.word) : null;
-        if (numericToolTip) {
-            this.decorations.numericToolTip = {
-                range: e.target.range,
-                options: {isWholeLine: false, hoverMessage: ['`' + numericToolTip + '`']}
-            };
-            this.updateDecorations();
-        }
-        if (this.settings.hoverShowAsmDoc === true) {
-            var asmToolTip = currentWord && currentWord.word ? getAsmToolTip(currentWord.word) : null;
-            if (asmToolTip) {
-                this.decorations.asmToolTip = {
+        if (currentWord && currentWord.word) {
+            var numericToolTip = getNumericToolTip(currentWord.word);
+            if (numericToolTip) {
+                this.decorations.numericToolTip = {
                     range: e.target.range,
-                    options: {
-                        isWholeLine: false,
-                        hoverMessage: [asmToolTip + " More information available in the context menu."]
-                    }
+                    options: {isWholeLine: false, hoverMessage: ['`' + numericToolTip + '`']}
                 };
                 this.updateDecorations();
             }
+
+            if (this.settings.hoverShowAsmDoc === true) {
+                getAsmInfo(currentWord.word).then(_.bind(function(response) {
```

Comment:
Might need a check here to ensure the opcode under consideration is still valid. If the mouse has moved since the request was sent we could end up showing a stale old request.